runtime/cudaq/platform/default/rest/helpers/oqc/OQCServerHelp.cpp

```diff
@@ -0,0 +1,298 @@
+
+  /// @brief Checks if a job is done based on the server's response to a job
+  /// retrieval request.
+  bool jobIsDone(ServerMessage &getJobResponse) override;
+
+  /// @brief Processes the server's response to a job retrieval request and
+  /// maps the results back to sample results.
+  cudaq::sample_result processResults(ServerMessage &postJobResponse) override;
+
+  std::string get_from_config(BackendConfig config, std::string key);
+  std::string get_from_config(BackendConfig config, std::string key,
+                              std::string default_return);
+};
+
+// Initialize the OQC server helper with a given backend configuration
+void OQCServerHelper::initialize(BackendConfig config) {
+  cudaq::info("Initializing OQC Backend.");
+  // Move the passed config into the member variable backendConfig
+  backendConfig = std::move(config);
+  // Set the necessary configuration variables for the OQC API
+  backendConfig["url"] = OQCServerHelper::get_from_config(config, "url");
+  backendConfig["version"] = "v0.3";
+  backendConfig["user_agent"] = "cudaq/0.3.0";
+  backendConfig["target"] = "Lucy";
+  backendConfig["qubits"] = 8;
+  // Retrieve the API key from the environment variables
+  backendConfig["email"] = OQCServerHelper::get_from_config(config, "email");
+  backendConfig["password"] =
+      OQCServerHelper::get_from_config(config, "password");
+  // Construct the API job path
+  backendConfig["job_path"] = "/tasks"; // backendConfig["url"] + "/tasks";
+}
+
+std::string OQCServerHelper::get_from_config(BackendConfig config,
+                                             std::string key,
+                                             std::string default_return) {
+  std::string output = OQCServerHelper::get_from_config(config, key);
+  if (output.empty()) {
+    return default_return;
+  }
+  return output;
+}
+
+std::string OQCServerHelper::get_from_config(BackendConfig config,
+                                             std::string key) {
+  auto iter = backendConfig.find(key);
+  if (iter != backendConfig.end()) {
+    return iter->second;
+  }
+  return "";
+}
+
+// Retrieve an environment variable
+std::string OQCServerHelper::getEnvVar(const std::string &key) const {
+  // Get the environment variable
+  const char *env_var = std::getenv(key.c_str());
+  // If the variable is not set, throw an exception
+  if (env_var == nullptr) {
+    throw std::runtime_error(key + " environment variable is not set.");
+  }
+  // Return the variable as a string
+  return std::string(env_var);
+}
+
+// Check if a key exists in the backend configuration
+bool OQCServerHelper::keyExists(const std::string &key) const {
+  return backendConfig.find(key) != backendConfig.end();
+}
+
+std::vector<std::string> OQCServerHelper::getJobID(int n) {
+  RestHeaders headers = OQCServerHelper::getHeaders();
+  nlohmann::json j;
+  std::vector<std::string> output;
+  for (int i = 0; i < n; ++i) {
+    nlohmann::json_v3_11_1::json response = client.post(
```

Comment:
Is there a reason to use `nlohmann::json_v3_11_1::json`, not just `nlohmann::json` (consistent with the signature of `cudaq::RestClient::post`? Specifying the internal ABI namespace of `nlohmann` would make the code less portable, IMHO.

Reply:
It looks like this was changed to `nlohmann::json` in the most recent commit, so I will mark this as resolved.